### ISSUE TITLE
Documentation missing

### ISSUE THREAD

**Issue by joschi27** [sirkris/Reddit.NET#95]
Hey, have you considered leaving information on the different classes and their respective methods? I've had 20 minutes just to find out how to display the text of a post, just because listing didn't sound like the thing i searched for.

Additionally, I can't find a way to fetch a post using it's permalink, is there a way to do this?

**Comment by sirkris** [comment]
Hi Joschi,

There are a number of code examples in the README, but you are right that there is no comprehensive documentation yet.  I hope to have that remedied soon.

In the meantime, everything is documented in the Intellisense, so that might help.  

As for looking up a post by permalink, you can do so by extracting the post id from the permalink string, prefacing it with "t3_" to convert it to a fullname, then passing that.

For example, let's say we have this permalink:

https://www.reddit.com/r/StillSandersForPres/comments/c8uu1x/want_to_see_bernie_at_the_top_of_the_polls_you/

The part we're looking for is "c8uu1x", the post id.  You can grab it with regex or whatever; it's always in the same place.  The rest of the permalink can be discarded.

Now we convert the post id to a post fullname by prefacing it with "t3_".  The resulting fullname string is "t3_c8uu1x".

Finally, we grab the post:

```c#
using Reddit;
using Reddit.Controllers;

...

Post post = reddit.Post(reddit.Models, "t3_c8uu1x").About();
```

This example is a self post, so you can access the post body like this:

```c#
SelfPost selfPost = reddit.SelfPost(reddit.Models, "t3_c8uu1x").About();

Console.WriteLine("Body: " 
    + selfPost.SelfText);
```

**Comment by joschi27** [comment]
Thank you very much for the explanation! I will try it later. And maybe you could add the code you juust showed me to the examples? 

 Well, I've gotten the library through nuget, and I don't really see what you mean with documented in intellisense.. The different classes haven't got any description for me, Visual Studio only opens the metadata and there are no comments there that would be displayed.

That would already tremendously help of course. Is there a way to display those?

The library itself is great though, made it a lot easier to use the api, so thank you for developing that!

**Comment by joschi27** [comment]
@sirkris Funnily enough, the nuget package doesn't have the overloaded method Post(Something, String)... Even though i can clearly see that the sourcecode does indeed have it. Weird, I guess i'll just compile the thing myself and won't use the nuget package. The comments should be visible also. Maybe look into this?

**Comment by sirkris** [comment]
It looks like the reason the Intellisense isn't showing up is because of something that the NuGet .nuspec file apparently requires that I didn't include.  I guess that would explain why so many people have been having difficulty finding their way around the library.

I should have this fixed in 1.3, as well.

**Comment by joschi27** [comment]
@sirkris Perfect! Thank you.

By the way, is there a way to download the subreddit icons? Or the upvote icons?

**Comment by sirkris** [comment]
Went ahead and added the new examples to the README:

[Use a Permalink to Retrieve a Reddit Post](https://github.com/sirkris/Reddit.NET/blob/develop/docs/examples/cs/Get%20Post%20From%20Permalink.md)

[Retrieve a LinkPost URL or SelfPost Body From a Reddit Post](https://github.com/sirkris/Reddit.NET/blob/develop/docs/examples/cs/Get%20LinkPost%20URL%20%26%20SelfPost%20Body%20From%20Post.md)

**Comment by sirkris** [comment]
Upvote icons are defined in the subreddit stylesheet, which can be accessed like this:

```c#
string stylesheet = reddit.Subreddit("marijuana").Stylesheet();
```

Unfortunately, the endpoint that uses requires the modconfig scope, which means the above call will fail unless you're a moderator of the subreddit in question.  As far as I know, it is not possible to retrieve the stylesheet via the API if you're not a moderator.

Accessing the subreddit icon is a lot easier:

```c#
reddit.Subreddit("marijuana").CommunityIcon
```

**Comment by joschi27** [comment]
Allright, thank you very much! I managed it by just using a html client and getting the node with the right classes, downloading the source, will change that though because that way is faster :) Thank you!

**Comment by sirkris** [comment]
I've added full reference documentation using Doxygen and merged it into 1.3.  Intellisense should now also work with NuGet installs.  Please feel free to re-open this ticket if you encounter any problems with either.
